File: jest/src/llm.ts
import { ChatOpenAI } from "@langchain/openai";
import { StringOutputParser } from "@langchain/core/output_parsers";
import {
    PromptTemplate,
} from "@langchain/core/prompts";

export class LLM {
    model: ChatOpenAI;

    constructor() {
        this.model = new ChatOpenAI({ model: "gpt-4o"});
    }

    public async invoke(prompt: string, variables: any): Promise<any> {
        const promptTemplate = PromptTemplate.fromTemplate(prompt);
        const formattedPrompt = await promptTemplate.format(variables);

        const chain = this.model.pipe(new StringOutputParser());
        return await chain.invoke(formattedPrompt);
    }
}

File: pytest/unitprompt/llm.py
from langchain_core.output_parsers import StrOutputParser
from langchain_core.prompts import PromptTemplate
from langchain_core.messages import HumanMessage
from langchain_openai import ChatOpenAI
from dotenv import load_dotenv

class LLM:
    def __init__(self):
        load_dotenv()
        self.model = ChatOpenAI(model="gpt-4o-mini")

    async def invoke(self, prompt: str, variables: dict) -> str:
        prompt_template = PromptTemplate.from_template(prompt)
        
        # Use the model to generate a response
        chain = prompt_template | self.model | StrOutputParser()
        return chain.invoke(variables)
